v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/SpannerToSourceDb.java

```diff
@@ -524,7 +528,10 @@ public static PipelineResult run(Options options) {
                     options.getSourceDbTimezoneOffset(),
                     ddl,
                     options.getShadowTablePrefix(),
-                    options.getSkipDirectoryName()));
+                    options.getSkipDirectoryName(),
+                    debugOptions.getNumberOfWorkerHarnessThreads() > 0
+                        ? debugOptions.getNumberOfWorkerHarnessThreads()
```

Comment:
I had a general question on how jdbcIO sets the pool size per worker, since it does not seem to be using getNumberOfWorkerHarnessThreads in the code that I see? We can maybe replicate whatever they do.

If this is giving improvements as per testing, we can take that as a followup.

Reply:
JDBCIO gives abstractions to provide the DataSourceProvider - the caller can create it based on their choice of the connection pool, size etc. Meaning there is no set way.  There is a default DBCP2 pool (PoolingDataSource ) - pool size is default to -1 not sure what DBCP2 interprets it as. I could not find in-house implementation where connection pool size is provided ( for instance CdcJdbcIO and DynamicJdbcIO,v2/jdbc-common/src/main/java/com/google/cloud/teleport/v2/auto/blocks/ReadFromJdbc.java)

I have changed the connection pool logic a bit: I have taken max number of workers configured and the max connections allowed per shard - took their division as the connection pool size for the worker. This is more intuitive - since it guarantees that it will never cross the max allowed connections.